controllers/cloud/internal/controller/cloudclient_controller.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	cloudv1 "github.com/labring/sealos/controllers/cloud/api/v1"
+	"github.com/labring/sealos/controllers/cloud/internal/controller/util"
+	cloud "github.com/labring/sealos/controllers/cloud/internal/manager"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// CloudClientReconciler reconciles a CloudClient object
+type CloudClientReconciler struct {
+	client.Client
+	Scheme     *runtime.Scheme
+	logger     logr.Logger
+	configPath string
+}
+
+//+kubebuilder:rbac:groups=cloud.sealos.io,resources=cloudclients,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=cloud.sealos.io,resources=cloudclients/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=cloud.sealos.io,resources=cloudclients/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the CloudClient object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+func (r *CloudClientReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.logger.Info("Enter CloudClientReconcile", "namespace:", req.Namespace, "name", req.Name)
+
+	r.logger.Info("Start the cloud module...")
+
+	var clusterSecret corev1.Secret
+	var config, err = util.ReadConfigFile(r.configPath, r.logger)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.logger.Info("Try to get the cloud secret resource...")
+	resource := util.NewImportanctResource(&clusterSecret, types.NamespacedName{Namespace: cloud.Namespace, Name: cloud.SecretName})
```

Comment:
何为 Importenct Resource?

Reply:
Some resources related to Licenses contain important information that needs to be stored persistently. If these resources are deleted, it could lead to unforeseen consequences. Therefore, related callback functions are triggered in GetImportanceResource to detect intrusion behavior, and appropriate actions will be taken when such behavior is discovered.